packages/steps/src/BuildWorkflow.ts

```diff
@@ -23,7 +23,9 @@ export class BuildWorkflow {
         try {
           await step.executeAsync();
         } catch (err: any) {
-          maybeError = err;
+          if (!maybeError) {
+            maybeError = err;
+          }
```

Comment:
I find it cleaner, but I'll let you decide

```suggestion
          maybeError = maybeError ?? err;
```